### PR TITLE
docs: update default-case with TypeScript info

### PR DESCRIPTION
**Pull request by Zamiell**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update

### CODE REVIEW

**Comment by Zamiell** [comment]
As a side note, I don't understand what the point of requiring the first commit message is match conventional commits is, since it looks like when PRs are squashed it uses the name of the PR instead of the name of the first commit, and then the automated stuff that the bot mentions are created from that. Is that just a bug in the bot?

**Comment by mdjermanovic** [comment]
> As a side note, I don't understand what the point of requiring the first commit message to match conventional commits is, since it looks like when PRs are squashed it uses the name of the PR instead of the name of the first commit, and then the automated stuff that the bot mentions are created from that. Is that just a bug in [eslint-github-bot](https://github.com/apps/eslint-github-bot)?

If there's only one commit, its message is the default for the squash commit message, so the bot is correct. When there are multiple commits, PR title is the default.

I believe GitHub now provides a configuration to always use PR title, so we could consider changing this.

**Comment by Zamiell** [comment]
Ah, thanks for the info. Yes, it does seem better to use the PR title when there is only one commit.

File: docs/src/rules/default-case.md [hunk 1]
```diff
@@ -137,3 +137,5 @@ switch(a) {
 ## When Not To Use It
 
 If you don't want to enforce a `default` case for `switch` statements, you can safely disable this rule.
+
+If you are using TypeScript, note that using this rule is actively harmful because it conflicts with the [`@typescript-eslint/switch-exhaustiveness-check`](https://typescript-eslint.io/rules/switch-exhaustiveness-check/) rule. In other words, some switch statements have a case for each of the distinct values in a union or enum. In these kinds of switch statements, having a `default` statement would prevent the `switch-exhaustiveness-check` rule from triggering when the union/enum is modified, which is the whole point of using the rule in the first place.
```
Comment (nzakas) [inline docs/src/rules/default-case.md:141]: This isn't actually about TypeScript, but rather a conflict with a plugin. As such, I don't think this is the right place to document this info. It should be documented on the `typescript-eslint` rule documentation.

I don't want to keep references to what other plugins are doing in the docs as it's very easy to get stale.

**Pull request closed without merging**